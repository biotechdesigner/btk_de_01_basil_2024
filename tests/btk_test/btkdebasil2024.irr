************************************************************************
pyfao56: FAO-56 Evapotranspiration in Python
Irrigation Data
Timestamp: 11/01/2023 11:28:37
************************************************************************
Comments: 2013 Cotton Dry
************************************************************************
Year-DOY  Depth     fw
2024-077  33.00   0.30
