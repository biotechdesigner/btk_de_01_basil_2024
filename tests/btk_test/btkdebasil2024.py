"""
########################################################################
The btkdebasil2024.py module contains a function to setup and run pyfao56
for the treatment in a 2024 basil greenhouse study at
Hohenheim, Germany.  The savefile and loadfile routines for Parameters,
Weather, and Irrigation classes are also tested.

The btkdebasil2024.py module contains the following:
    run - function to setup and run pyfao56 for the
    treatment in a 2024 basil greenhouse study at Hohenheim, Germany

03/11/2024 Scripts developed for running pyfao56 for 2024 basil data
########################################################################
"""

import pyfao56 as fao
import os

def run():
    """Setup and run pyfao56 for a 2024 basil greenhouse study"""

    #Get the module directory
    module_dir = os.path.dirname(os.path.abspath(__file__))

    #Specify the model parameters
    par = fao.Parameters(comment = '2024 Basil')
    par.Kcbini = 0.40
    par.Kcbmid = 1.10
    par.Kcbend = 1.05
    par.Lini = 15
    par.Ldev = 30
    par.Lmid = 20
    par.Lend = 20
    par.hini = 0.1
    par.hmax = 0.8
    par.thetaFC = 0.14
    par.thetaWP = 0.060
    par.theta0 = 0.100
    par.Zrini = 0.25
    par.Zrmax = 0.80
    par.pbase = 0.40
    par.Ze = 0.1143
    par.REW = 8.0
    par.savefile(os.path.join(module_dir,'btkdebasil2024.par'))
    par.loadfile(os.path.join(module_dir,'btkdebasil2024.par'))

    #Specify the weather data
    wth = fao.Weather(comment = '2024 basil')
    wth.loadfile(os.path.join(module_dir,'btkdebasil2024'))
    wth.savefile(os.path.join(module_dir,'btkdebasil2024'))
    wth.loadfile(os.path.join(module_dir,'btkdebasil2024'))

    #Specify the irrigation schedule
    """
    Atributes
    -------
        index - Year and day of year as string ('yyyy-ddd')
        columns - ['Depth','fw']
            Depth - Irrigation depth (mm)
            fw - fraction of soil surface wetted (FAO-56 Table 20)
            
    Methods
    -------
    savefile(filepath='pyfao56.irr')
        Save irrigation data to a file
    loadfile(filepath='pyfao56.irr')
        Load irrigation data from a file
    addevent(year,doy,depth,fw)
        Add an irrigation event to self.idata
    customload()
        Users can override for custom loading of irrigation data.
    """
    irr = fao.Irrigation(comment = '2024 basil')
    irr.addevent(2013, 115, 33.0, 0.5)
    irr.savefile(os.path.join(module_dir,'btkdebasil2024.irr'))
    irr.loadfile(os.path.join(module_dir,'btkdebasil2024.irr'))

    #Run the model
    mdl = fao.Model('2013-113','2013-312', par, wth, irr=irr,
                    comment = '2024 basil')
    mdl.run()
    print(mdl)
    mdl.savefile(os.path.join(module_dir,'btkdebasil2024.out'))
    mdl.savesums(os.path.join(module_dir,'btkdebasil2024.sum'))

if __name__ == '__main__':
    run()
