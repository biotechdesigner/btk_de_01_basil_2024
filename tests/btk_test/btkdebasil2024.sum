************************************************************************
pyfao56: FAO-56 Evapotranspiration in Python
Seasonal Water Balance Summary
Timestamp: 03/18/2024 13:45:22
Simulation start date: 03/17/2024
Simulation end date: 03/17/2024
All values expressed in mm.
************************************************************************
Comments: 2024 basil
************************************************************************
   1.284 : ETref
   0.513 : ETc
   0.513 : ETcadj
   0.000 : E
   0.513 : T
  22.487 : DP
  33.000 : Irrig
   0.000 : Rain
   0.000 : Dr_ini
   0.000 : Dr_end
 -99.999 : Drmax_ini
 -99.999 : Drmax_end
