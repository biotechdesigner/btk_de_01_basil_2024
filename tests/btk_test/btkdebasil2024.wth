************************************************************************
pyfao56: FAO-56 Evapotranspiration in Python
Weather Data
Timestamp: 03/18/2024 13:45:22
************************************************************************
Comments:
************************************************************************
           S Reference crop - Short ('S') or Tall ('T')
 389.0000000 Weather station elevation (z) (m)
  48.7111700 Weather station latitude (decimal degrees)
   2.0000000 Wind speed measurement height (m)

Daily weather data:
Year-DOY   Srad   Tmax   Tmin   Vapr   Tdew  RHmax  RHmin  Wndsp   Rain  ETref   MorP
2024-077  10.29  13.30   4.30    NaN    NaN  98.90  58.80   0.50   0.00    NaN      M