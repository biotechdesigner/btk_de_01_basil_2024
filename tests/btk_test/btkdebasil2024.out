************************************************************************
pyfao56: FAO-56 Evapotranspiration in Python
Output Data
Timestamp: 03/18/2024 13:45:22
Simulation start date: 03/17/2024
Simulation end date: 03/17/2024
Soil method: D - Default FAO-56 homogenous soil bucket approach
************************************************************************
Comments: 2024 basil
************************************************************************
Year-DOY  Year  DOY  DOW      Date  ETref  tKcb   Kcb     h Kcmax    fc    fw   few      De    Kr    Ke      E     DPe    Kc    ETc     TAW TAWrmax    TAWb    Zr     p     RAW    Ks Kcadj ETcadj      T      DP    Dinc      Dr     fDr   Drmax  fDrmax      Db     fDb   Irrig    Rain  Year  DOY  DOW      Date
2024-077  2024  077  Sun  03/17/24  1.284 0.400 0.400 0.100 1.166 0.000 0.300 0.300   0.000 0.000 0.000  0.000  97.427 0.400  0.513  20.000 -99.999 -99.999 0.250 0.579  11.589 1.000 0.400  0.513  0.513  22.487 -99.999   0.000   0.000 -99.999 -99.999 -99.999 -99.999  33.000   0.000  2024  077  Sun  03/17/24